packages/manager/modules/hub/src/components/support/controller.js

```diff
@@ -0,0 +1,46 @@
+import get from 'lodash/get';
+import isEmpty from 'lodash/isEmpty';
+import { MAX_TICKETS_TO_DISPLAY } from './constants';
+import SupportTicket from './ticket.class';
+
+export default class ManagerHubSupportCtrl {
+  /* @ngInject */
+  constructor($http, $q, RedirectionService) {
+    this.$http = $http;
+    this.$q = $q;
+    this.RedirectionService = RedirectionService;
+  }
+
+  $onInit() {
+    this.isLoading = true;
+    return this.$q
+      .when(this.tickets ? this.tickets : this.fetchTickets())
+      .then((tickets) => {
+        if (Array.isArray(tickets)) {
+          this.tickets = tickets
+            .slice(0, MAX_TICKETS_TO_DISPLAY)
+            .map((ticket) => new SupportTicket(ticket));
+        }
+      })
+      .then(() => isEmpty(this.tickets) && this.fetchGuide())
+      .then((guideURL) => {
+        this.guideURL = guideURL;
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
+
+  fetchGuide() {
+    return this.RedirectionService.getGuideURL('home');
+  }
+
+  fetchTickets() {
```

Comment:
I'm quite puzzled about this, on one hand I found a good idea that the component fetches its data if none is provided but on the other hand I'm wondering if it is not doing too much then handling that logic

Reply:
It's doing too much but it may be premature to have the alternative, which is a generic way to inject data from the API into our components